src/PolygonDraw/Map.test.tsx

```diff
@@ -1,174 +1,79 @@
-import { LeafletMouseEvent } from 'leaflet';
 import React from 'react';
 import { shallow, ShallowWrapper } from 'enzyme';
-import { MAP } from '../constants';
+import { LeafletMouseEvent } from 'leaflet';
 
-import { subtractCoordinates, createLeafletLatLngFromCoordinate, getPolygonEdges, getCenterCoordinate } from '../helpers';
+import { MAP } from '../constants';
+import {
+    subtractCoordinates,
+    createLeafletLatLngFromCoordinate,
+    getPolygonEdges,
+    getCenterCoordinate
+} from '../helpers';
 import { Props, BaseMap, State } from './Map';
 import { MOCK_POLYGON } from '../mockPolygon';
 import { PolygonVertex } from './PolygonVertex';
-import { actions } from './actions';
-import { Map } from '../leaflet/Map';
 
 describe('Map component', () => {
-    const OPEN_POLYGON = [MOCK_POLYGON[0]];
     const coordinate = getCenterCoordinate(MOCK_POLYGON[10], MOCK_POLYGON[12]);
     const event = { latlng: createLeafletLatLngFromCoordinate(coordinate) } as LeafletMouseEvent;
 
     let initialProps: Props;
     let wrapper: ShallowWrapper<Props, State, BaseMap>;
 
     beforeEach(() => {
-
         initialProps = {
-            polygonCoordinates: MOCK_POLYGON,
+            activePolygonIndex: 0,
+            polygonCoordinates: [MOCK_POLYGON],
             boundaryPolygonCoordinates: MOCK_POLYGON,
             selection: new Set<number>(),
             editable: true,
-            dispatch: jest.fn(),
             initialCenter: MAP.DEFAULT_CENTER,
-            initialZoom: MAP.DEFAULT_ZOOM
+            initialZoom: MAP.DEFAULT_ZOOM,
+            isPolygonClosed: true,
+            addPoint: jest.fn(),
+            addPointToEdge: jest.fn(),
+            addPointsToSelection: jest.fn(),
+            deselectAllPoints: jest.fn(),
+            removePointFromSelection: jest.fn(),
+            selectPoints: jest.fn(),
+            selectAllPoints: jest.fn(),
+            moveSelectedPoints: jest.fn(),
+            deletePolygonPoints: jest.fn()
         };
     });
 
-    describe('WHEN polygon is NOT disabled', () => {
+    afterEach(() => {
+        jest.resetAllMocks();
+    });
 
+    describe('WHEN polygon is NOT disabled', () => {
         it('should NOT enable pen tool', () => {
             wrapper = shallow(<BaseMap {...initialProps} editable={false} />);
             expect(wrapper.state().isPenToolActive).toBeFalsy();
         });
-
-        describe('when user moves mouse on map', () => {
-            it('should not track mouse position if pen tool is disabled', () => {
-                wrapper = shallow(<BaseMap {...initialProps} polygonCoordinates={OPEN_POLYGON} />);
-                const MapElement = wrapper.find(Map);
-
-                expect(wrapper.state().newPointPosition).toEqual(null);
-                // @ts-ignore
-                MapElement.prop('onmousemove')(event);
-
-                setTimeout(() => {
```

Comment:
We removed a lot of these tests due to setTimeout. Eventually we should add real test that work for this. I'll create a followup to write integration tests for the PolygonDraw component.